src/java/picard/sam/PositionBasedDownsampleSam.java

```diff
@@ -0,0 +1,295 @@
+                "<it>Note 3</it>: Downsampling twice with this program is not supported." +
+                "<it>Note 4</it>: You should MarkDuplicates after downsampling." +
+                "\n" +
+                "Finally, the code has been designed to simulate sequencing less as accurately as possible, not for getting an exact downsample fraction. " +
+                "In particular, since the reads may be distributed non-evenly within the lanes/tiles, the resulting downsampling percentage will not be accurately" +
+                "determined by the input argument PROBABILITY.",
+        usageShort = "Down-sample a SAM or BAM file to retain a subset of the reads based on the reads location in each tile in the flowcell.",
+        programGroup = SamOrBam.class
+)
+public class PositionBasedDownsampleSam extends CommandLineProgram {
+
+    @Option(shortName = StandardOptionDefinitions.INPUT_SHORT_NAME, doc = "The input SAM or BAM file to downsample.")
+    public File INPUT;
+
+    @Option(shortName = StandardOptionDefinitions.OUTPUT_SHORT_NAME, doc = "The output, downsampled, SAM or BAM file to write.")
+    public File OUTPUT;
+
+    @Option(shortName = StandardOptionDefinitions.PROBABILITY_SHORT_NAME, doc = "The (approximate) probability of keeping a read, between 0 and 1.")
+    public double PROBABILITY = 1;
+
+    @Option(doc = "Stop after processing N reads, mainly for debugging.")
+    public long STOP_AFTER = 0;
+
+    @Option(doc = "Determines whether the duplicate tag should be reset since the downsampling requires re-marking duplicates.")
+    public boolean RESET_DUPLICATE_FLAG = true;
+
+    private final Log log = Log.getInstance(PositionBasedDownsampleSam.class);
+
+    private CircleSelector selector = new CircleSelector(PROBABILITY);
+
+    private PhysicalLocation opticalDuplicateFinder;
+    private long total = 0;
+    private long kept = 0;
+
+    //max-position in tile as a function of tile. We might need to
+    //look per-readgroup, but at this point I'm making the assumptions that I need to downsample a
+    //sample where all the readgroups came from the same type of flowcell.
+    private Map<Short, Coord> tileMaxCoord;
+
+    final Map<Short, Histogram<Short>> xPositions = new HashMap<Short, Histogram<Short>>();
+    final Map<Short, Histogram<Short>> yPositions = new HashMap<Short, Histogram<Short>>();
+
+    public static void main(final String[] args) {
+        new PositionBasedDownsampleSam().instanceMainWithExit(args);
+    }
+
+    @Override
+    protected String[] customCommandLineValidation() {
+        final List<String> errors = new ArrayList<String>();
+
+        if (PROBABILITY < 0 || PROBABILITY > 1) {
+            errors.add("PROBABILITY must be a value between 0 and 1, found: " + PROBABILITY);
+        }
+
+        IOUtil.assertFileIsReadable(INPUT);
```

Comment:
Validate the values of the command line parameters, like you do, especially if there are dependencies between them.  It is not wrong to put file existence checking their, it is just more useful if they are in the ```doWork``` method due to my reasons above, and for consistency in the code (legal precedence?).